### PR TITLE
Switch CentOS 7 to rh-postgresql10 SCL

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

File: puppet/modules/slave/manifests/postgresql.pp [hunk 1]
```diff
@@ -1,22 +1,38 @@
...
+        version              => '10',
+        client_package_name  => 'rh-postgresql10-postgresql-syspaths',
+        server_package_name  => 'rh-postgresql10-postgresql-server-syspaths',
+        contrib_package_name => 'rh-postgresql10-postgresql-contrib-syspaths',
+        service_name         => 'postgresql',
+        datadir              => '/var/opt/rh/rh-postgresql10/lib/pgsql/data',
+        confdir              => '/var/opt/rh/rh-postgresql10/lib/pgsql/data',
+        bindir               => '/usr/bin',
+      }
+      file { '/usr/bin/pg_config':
```
Comment (mmoll) [inline puppet/modules/slave/manifests/postgresql.pp]: It seems there's no `rh-postgresql10-postgresql-devel-syspaths`, which would bring this..

File: puppet/modules/slave/manifests/init.pp [hunk 2]
```diff
@@ -101,8 +101,11 @@
     'postgresql-dev':
```
Comment (ekohl) [inline puppet/modules/slave/manifests/init.pp]: Should be use `postgresql::lib::devel` from puppetlabs-postgresql?

File: puppet/modules/slave/manifests/postgresql.pp [hunk 3]
```diff
@@ -1,22 +1,38 @@
 # @api private
 class slave::postgresql {
-  Postgresql_psql {
-    cwd => '/',
-  }
+  # only CentOS slaves are used to run unit tests
+  if $::osfamily == 'RedHat' {
+    if ::operatingsystemmajrelease == '7' {
+      include scl
```
Comment (ekohl) [inline puppet/modules/slave/manifests/postgresql.pp]: Why is this needed? The whole point of the `-syspaths` packages is that you don't need to explicitly enable them.

File: puppet/modules/slave/manifests/postgresql.pp [hunk 4]
```diff
@@ -1,22 +1,38 @@
...
-  # Simple known user/pass that will allow Jenkins to create its
-  # own databases when required
-  postgresql::server::role { 'foreman':
-    password_hash => postgresql_password('foreman', 'foreman'),
-    superuser     => true,
-    login         => true,
-  }
+      $non_scl_rpms = ['postgresql', 'postgresql-server', 'postgresql-devel', 'postgresql-libs']
+      $non_scl_rpms.each |String $non_scl_rpm| {
+        package { "non_scl_${non_scl_rpm}":
```
Comment (ekohl) [inline puppet/modules/slave/manifests/postgresql.pp]: You can also specify it at once. Probably also good to ensure it happens before postgresql is ran.
```ruby
package { $non_scl_rpms:
  ensure => absent,
  before => Class['postgresql::globals'],
}
```

File: puppet/modules/slave/manifests/postgresql.pp (hunk 1 above)
Comment (ekohl) [inline puppet/modules/slave/manifests/postgresql.pp]: I think that if you install `rh-postgresql10-syspaths` this shouldn't be needed.

File: puppet/modules/slave/manifests/postgresql.pp (hunk 4 above)
Comment (mmoll) [reply puppet/modules/slave/manifests/postgresql.pp]: Nope...
```
Error: Evaluation Error: Error while evaluating a Resource Statement, Duplicate declaration: Package[postgresql-devel] is already declared at (file: /root/foreman-infra/puppet/modules/postgresql/manifests/lib/devel.pp, line: 21); cannot redeclare (file: /root/foreman-infra/puppet/modules/slave/manifests/postgresql.pp, line: 19) (file: /root/foreman-infra/puppet/modules/slave/manifests/postgresql.pp, line: 19, column: 1) on node localhost.localdomain
```

File: puppet/modules/slave/manifests/init.pp (hunk 2 above)
Comment (mmoll) [reply puppet/modules/slave/manifests/init.pp]: If it work with SCLs, yes... setting the devel_package_name seems not to suffice..

File: puppet/modules/slave/manifests/postgresql.pp (hunk 3 above)
Comment (mmoll) [reply puppet/modules/slave/manifests/postgresql.pp]: At least in my test VM SCLs are not enabled by default, so nothing can get installed.

File: puppet/modules/slave/manifests/postgresql.pp (hunk 1 above)
Comment (mmoll) [reply puppet/modules/slave/manifests/postgresql.pp]: Nope...
```
[root@localhost ~]# rpm -q rh-postgresql10-syspaths
rh-postgresql10-syspaths-3.1-1.el7.x86_64
[root@localhost ~]# which pg_config
/usr/bin/which: no pg_config in (/sbin:/bin:/usr/sbin:/usr/bin:/opt/puppetlabs/bin:/usr/local/rvm/bin)

```

File: puppet/modules/slave/manifests/postgresql.pp (hunk 3 above)
Comment (ekohl) [reply puppet/modules/slave/manifests/postgresql.pp]: Perhaps it's easier to just install `centos-release-scl-rh` then to get the repos?

**Comment by ekohl** [comment]
Btw, in the SCL config we still build with the old libs. It should be possible to use the non-SCL libs but SCL server since the wire protocol hasn't changed. Perhaps that's an easier goal and more realistic of the actual deployment?

File: puppet/modules/slave/manifests/postgresql.pp (hunk 3 above)
Comment (mmoll) [reply puppet/modules/slave/manifests/postgresql.pp]: I think I need to test all that also on RHEL first, as we do have the AWS slaves on RHEL.

Comment (ekohl) [reply puppet/modules/slave/manifests/postgresql.pp]: On RHEL I think it's a separate subscription. Not entirely sure.

**Comment by mmoll** [comment]
This should work now... on the AWS slaves I installed the `centos-release-scl-rh` RPM manually.

Locally I had postgresql not starting up when SELinux was enabled, but frankly I'm not willing to care about this. :broken_heart:

**Comment by ehelms** [comment]
Might as well upgrade to 12 if thats easy enough, thats what we are moving nightly to now that they released the SCL for PG12.

**Comment by ekohl** [comment]
@ehelms the AWS machines are RHEL, not CentOS - do you know if we can use the RH SCLs rather than CentOS on them?

**Comment by mmoll** [comment]
I'm leaving this to the RH experts, because ENOTIME :crying_cat_face:
